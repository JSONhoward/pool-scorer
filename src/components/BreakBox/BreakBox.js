import React from 'react'
import styled from 'styled-components'
import {FaLevelDownAlt } from 'react-icons/fa'

const BreakBoxStyled = styled.div`
display: flex;
opacity: ${props => props.gameOver ? '0' : '1'};
justify-content: space-between;
width: 220px;
font-size: 2rem;
`

const ArrowLeft = styled.div`
display: flex;
visibility: ${props => props.player1 ? 'visible' : 'hidden'};
align-items: flex-end;
justify-content: center;
flex: 1;
transform: rotateX(180deg) rotateY(180deg);

`

const ArrowRight = styled.div`
display: flex;
visibility: ${props => props.player1 ? 'visible' : 'hidden'};
align-items: flex-end;
justify-content: center;
flex: 1;
transform: rotateX(180deg);
`

const BreakBox = ({player1, gameOver}) => {
    return (
        <BreakBoxStyled gameOver={gameOver}>
            <ArrowLeft player1={player1}><FaLevelDownAlt /></ArrowLeft>
            <p>Break</p>
            <ArrowRight player1={!player1}><FaLevelDownAlt /></ArrowRight>
        </BreakBoxStyled>
    )
}

export default BreakBox
