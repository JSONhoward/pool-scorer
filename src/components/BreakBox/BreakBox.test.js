import React from 'react'
import {create} from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import BreakBox from './BreakBox'

describe('BreakBox Tests', () => {
    test('Snapshot Test', () => {
        const tree = create(<BreakBox />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Renders Correctly', () => {
        const component = create(<BreakBox />).root

        expect(component.children.length).toEqual(1)
    })

    test('Receives Props', () => {
        const component = create(<BreakBox player1={true} gameOver={false} />).root

        console.log(component.children.length)
        expect(component.findByProps({player1: true})).toBeTruthy()
        expect(component.findByProps({gameOver: false})).toBeTruthy()
    })
})