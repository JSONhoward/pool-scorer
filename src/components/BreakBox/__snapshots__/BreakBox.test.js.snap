// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreakBox Tests Snapshot Test 1`] = `
<div
  className="sc-AxjAm hnPlRi"
>
  <div
    className="sc-AxirZ hiwWPf"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.553 392.331L209.587 504.334c-9.485 10.214-25.676 10.229-35.174 0L70.438 392.331C56.232 377.031 67.062 352 88.025 352H152V80H68.024a11.996 11.996 0 0 1-8.485-3.515l-56-56C-4.021 12.926 1.333 0 12.024 0H208c13.255 0 24 10.745 24 24v328h63.966c20.878 0 31.851 24.969 17.587 40.331z"
      />
    </svg>
  </div>
  <p>
    Break
  </p>
  <div
    className="sc-AxiKw PlJov"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.553 392.331L209.587 504.334c-9.485 10.214-25.676 10.229-35.174 0L70.438 392.331C56.232 377.031 67.062 352 88.025 352H152V80H68.024a11.996 11.996 0 0 1-8.485-3.515l-56-56C-4.021 12.926 1.333 0 12.024 0H208c13.255 0 24 10.745 24 24v328h63.966c20.878 0 31.851 24.969 17.587 40.331z"
      />
    </svg>
  </div>
</div>
`;
